builder/amazon/common/step_run_source_instance.go

```diff
@@ -144,6 +147,23 @@ func (s *StepRunSourceInstance) Run(ctx context.Context, state multistep.StateBa
 		runOpts.CreditSpecification = &ec2.CreditSpecificationRequest{CpuCredits: &creditOption}
 	}
 
+	// Set Metadata defaults if not defined at all
+	if s.HttpEndpoint == "" {
+		s.HttpEndpoint = "enabled"
+	}
+
+	if s.HttpTokens == "" {
+		s.HttpTokens = "optional"
+	}
+
+	if s.HttpPutResponseHopLimit == 0 {
+		s.HttpPutResponseHopLimit = 1
+	}
```

Comment:
You don't need this here anymore cuz it should be done on `Prepare`. This way you don't need to set the defaults everywhere you use it. 
```suggestion
```